src/main/java/io/aiven/kafka/connect/common/output/jsonwriter/JsonLinesOutputStreamWriter.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.output.jsonwriter;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.Map;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import io.aiven.kafka.connect.common.output.OutputStreamWriter;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+
+class JsonLinesOutputStreamWriter implements OutputStreamWriter {
+
+    private final Map<String, OutputFieldBuilder> fieldBuilders;
+    private static final byte[] RECORD_SEPARATOR = "\n".getBytes(StandardCharsets.UTF_8);
+    private final ObjectMapper objectMapper;
+
+    JsonLinesOutputStreamWriter(final Map<String, OutputFieldBuilder> fieldBuilders) {
+        this.fieldBuilders = fieldBuilders;
+        this.objectMapper = new ObjectMapper();
+        objectMapper.setNodeFactory(JsonNodeFactory.withExactBigDecimals(true));
+    }
+
+    @Override
+    public void startWriting(final OutputStream outputStream) throws IOException {
```

Comment:
lets use empty implementation on `OutputStreamWriter` level, so you do not need to override it here. And I think I misunderstand something, are you sure that we need kind of state machine for stream?

Reply:
> And I think I misunderstand something, are you sure that we need kind of state machine for stream?

I might misunderstand a question. An idea is that `OutputStreamWriter` is a old class  `OutputStream` and nothing change for it expect a necessity to include `startWriting` for a classic JSON.  But  there is this new `AutoClosable` class that provides kind of simpler interface to a user and hide a necessity to call `startWriting` and `endWriting`.  Also, it makes sure that outputStream is always there - higher level object take care of it.


Of course, it is possible, to merge `OutputStreamWriter` and `OutputStream` to a one class. But in this case Json and JsonL become VERY similar (not an crime, of course) + This builder stuff. I'm afraid this make this class even bigger and harder to read. So, I tried to separate responsibilities a bit to make `OutputStreamWriter` class focus just on a format differences so it should be easier to introduce a new format.